### ISSUE TITLE
[4.0] clean install db errors

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#28530]
I just did a clean install in a new database.

Expected behaviour is that the admin does not show any database issues but 

1. it says there is one problem
One Problem
```
Table 'kh8be_user_notes' has the wrong type or attributes for column 'modified_user_id' with type int(10) unsigned. (From file 4.0.0-2018-06-26.sql.)
Table 'kh8be_finder_filters' has the wrong type or attributes for column 'created_by' with type int(10) unsigned. (From file 4.0.0-2018-07-29.sql.)
Table 'kh8be_finder_links' has the wrong type or attributes for column 'state' with type int(5). (From file 4.0.0-2018-07-29.sql.)
Table 'kh8be_finder_links' has the wrong type or attributes for column 'access' with type int(5). (From file 4.0.0-2018-07-29.sql.)
Table 'kh8be_extensions' has the wrong type or attributes for column 'protected' with type tinyint(3). (From file 4.0.0-2020-03-25.sql.)
130 database changes were checked.
227 database changes did not alter table structure and were skipped.
```

2. Clicking on "update structure" just returns with the notice "There are tables not up to date!" and nothing has been fixed

![image](https://user-images.githubusercontent.com/1296369/78119135-decc3100-73ff-11ea-886d-80c0264ed0f0.png)

**Comment by SharkyKZ** [comment]
See PR https://github.com/joomla/joomla-cms/pull/28501 for staging.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28530">issues.joomla.org/tracker/joomla-cms/28530</a>.</sub>

**Comment by brianteeman** [comment]
Reopening as this has to be fixed in [4.0] with waiting for a j3 pr to be merged and then go in n the queue for merging.

**Comment by richard67** [comment]
@brianteeman Would be nice if you could test my PR #28501 . Is same as you have tested once for J4.

**Comment by wilsonge** [comment]
marked as release blocker for j4 - so we can ensure that the pr bubbles up from staging

**Comment by richard67** [comment]
Well if PR #28501 shall bubble up in time, someone should test it. It needs only one more test.

**Comment by richard67** [comment]
Closing as solved with PR #29010 . I have just tested it.

@brianteeman Feel free to re-open if you can't confirm this.
